server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -596,6 +597,52 @@ private ClusterState applyCreateIndexRequestWithExistingMetadata(final ClusterSt
         return Collections.singletonMap(MapperService.SINGLE_MAPPING_NAME, finalMappings);
     }
 
+    /**
+     * Add the objects in the second map to the first, where the keys in the {@code second} map have
+     * higher predecence and overwrite the keys in the {@code first} map. In the event of a key with
+     * a dot in it (ie, "foo.bar"), the keys are treated as only the prefix counting towards
+     * equality. If the {@code second} map has a key such as "foo", all keys starting from "foo." in
+     * the {@code first} map are discarded.
+     */
+    static Map<String, Object> mergeIgnoringDots(Map<String, Object> first, Map<String, Object> second) {
```

Comment:
If I understand this correctly it can be simplified:
```java
static Map<String, Object> mergeIgnoringDots(Map<String, Object> first, Map<String, Object> second) {
    Objects.requireNonNull(first, "merging requires two non-null maps but the first map was null");
    Objects.requireNonNull(second, "merging requires two non-null maps but the second map was null");
    Map<String, Object> results = new HashMap<>(first);
    Set<String> prefixes = second.keySet().stream().map(MetadataCreateIndexService::prefix).collect(Collectors.toSet());
    results.keySet().removeIf(k -> prefixes.contains(prefix(k)));
    results.putAll(second);
    return results;
}

private static String prefix(String s) {
    return s.split("\\.", 2)[0];
}
```

Reply:
Sounds good, that works and passes the tests, updated the function.